src/DynamoCore/Models/NodeModel.cs

```diff
@@ -1210,30 +1210,32 @@ private void p_PortConnected(PortModel port, ConnectorModel connector)
         {
             ValidateConnections();
 
-            if (port.PortType == PortType.Input)
-            {
-                int data = InPorts.IndexOf(port);
-                PortModel startPort = connector.Start;
-                int outData = startPort.Owner.OutPorts.IndexOf(startPort);
-                ConnectInput(data, outData, startPort.Owner);
-                startPort.Owner.ConnectOutput(outData, data, this);
-                OnConnectorAdded(connector);
+            if (port.PortType != PortType.Input) return;
 
-                // OnNodeModified();
-            }
+            var data = InPorts.IndexOf(port);
+            var startPort = connector.Start;
+            var outData = startPort.Owner.OutPorts.IndexOf(startPort);
+            ConnectInput(data, outData, startPort.Owner);
+            startPort.Owner.ConnectOutput(outData, data, this);
+            OnConnectorAdded(connector);
+
+            MarkNodeAsModified();
+            OnNodeModified();
```

Comment:
But maybe you want to be sure that an overriding node doesn't prevent this from being hit?